packages/enzyme/src/ReactWrapper.js

```diff
@@ -104,19 +105,24 @@ class ReactWrapper {
         throw new TypeError('ReactWrapper can only wrap valid elements');
       }
 
-      privateSet(this, UNRENDERED, nodes);
       const renderer = adapter.createRenderer({ mode: 'mount', ...options });
+      if (options.wrappingComponent && typeof renderer.getWrappingComponentRenderer !== 'function') {
+        throw new Error('your adapter does not support `wrappingComponent`. Try upgrading it!');
+      }
       privateSet(this, RENDERER, renderer);
+      privateSet(this, WRAPPING_COMPONENT_RENDERER, renderer.getWrappingComponentRenderer
+        ? renderer.getWrappingComponentRenderer() : null);
       renderer.render(nodes, options.context);
       privateSet(this, ROOT, this);
       privateSetNodes(this, this[RENDERER].getNode());
     } else {
-      privateSet(this, UNRENDERED, null);
       privateSet(this, RENDERER, root[RENDERER]);
+      privateSet(this, WRAPPING_COMPONENT_RENDERER, root[WRAPPING_COMPONENT_RENDERER]);
       privateSet(this, ROOT, root);
       privateSetNodes(this, nodes);
       privateSet(this, ROOT_NODES, root[NODES]);
     }
+    privateSet(this, UNRENDERED, nodes);
```

Comment:
`UNRENDERED` needs to be set to call `.setProps()` on the `wrappingComponent`. It didn't seem like there was any harm in setting it to `nodes` regardless of `!root`. (All the tests still passed.)